File: 2358-number-of-ways-to-split-array/2358-number-of-ways-to-split-array.java
class Solution {
    public int waysToSplitArray(int[] nums) {
        long total_sum = 0;
        for(int i=0;i<nums.length;i++){
            total_sum +=nums[i];
        }
        int count = 0;
        long present_sum = 0;
        for(int i=0;i<nums.length-1;i++){
            present_sum += nums[i];
            long remaining = total_sum-present_sum;
            if(present_sum >= remaining){
                count++;
            }
        }
        return count;
    }
}

File: 2358-number-of-ways-to-split-array/2358-number-of-ways-to-split-array.py
class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:
        total_sum = 0
        for num in nums:
            total_sum +=num
        count = 0
        present_sum = 0
        for i in range(0,len(nums)-1):
            present_sum +=nums[i]
            remaining_sum = total_sum - present_sum
            if present_sum >= remaining_sum:
                count+=1
        return count